packages/next/build/webpack-config.ts

```diff
@@ -2062,8 +2062,7 @@ export default async function getBaseWebpackConfig(
           dev,
           sriEnabled: !dev && !!config.experimental.sri?.algorithm,
           hasFontLoaders: !!config.experimental.fontLoaders,
-          allowMiddlewareResponseBody:
-            !!config.experimental.allowMiddlewareResponseBody,
+          allowMiddlewareResponseBody: !!config.allowMiddlewareResponseBody,
```

Comment:
Lets handle these similar to https://github.com/vercel/next.js/pull/44194#discussion_r1053493981